EventListener/RateLimitAnnotationListener.php

```diff
@@ -100,19 +103,35 @@ public function onKernelController(FilterControllerEvent $event)
         }
 
         // When we exceeded our limit, return a custom error response
-        if ($rateLimitInfo->getCalls() > $rateLimitInfo->getLimit()) {
+        if (!$rateLimitInfo->isBlocked() && $rateLimitInfo->getCalls() > $rateLimitInfo->getLimit()) {
+            $this->rateLimitService->setBlock(
+                $rateLimitInfo,
+                $rateLimit->getBlockPeriod() > 0 ? $rateLimit->getBlockPeriod() : $rateLimit->getPeriod()
```

Comment:
I think If the block period isn't set, there should be no extended block period, as in how system works at the moment.